hack/tools/tilt-prepare/main.go

```diff
@@ -0,0 +1,486 @@
+}
+
+func tiltResources() error {
+	tasks := map[string]taskFunction{}
+
+	// If required, all the task to install cert manager.
+	// NOTE: strictly speaking cert-manager is not a resource, however it is a dependency for most of the actual resources
+	// and managing this is the same task group gives the maximum benefits in terms of reducing the total elapsed time.
+	if *certManagerFlag {
+		tasks["cert-manager-cainjector"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-cainjector:%s", config.CertManagerDefaultVersion))
+		tasks["cert-manager-webhook"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-webhook:%s", config.CertManagerDefaultVersion))
+		tasks["cert-manager-controller"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-controller:%s", config.CertManagerDefaultVersion))
+		tasks["cert-manager"] = certManagerTask()
+	}
+
+	// Add a kustomize task for each name/path defined using the --kustomize-build flag.
+	// NOTE
+	for _, k := range *kustomizeBuildsFlag {
+		values := strings.Split(k, ":")
+		if len(values) != 2 {
+			return fmt.Errorf("[resources] failed to parse --kustomize-build flag %s: value should be in the form of name:path:debug[:[var=value][;]]", k)
+		}
+		name := values[0]
+		path := values[1]
+		tasks[name] = kustomizeTask(path, fmt.Sprintf("%s.yaml", name))
+	}
+
+	// Add a provider task for each name/path defined using the --provider flag.
+	for _, p := range *providersBuildsFlag {
+		pValues := strings.Split(p, ":")
+		if len(pValues) != 2 && len(pValues) != 3 {
+			return fmt.Errorf("[resources] failed to parse --provider flag %s: value should be in the form of name:path[:debug]", p)
+		}
+		name := pValues[0]
+		path := pValues[1]
+		debug := false
+		if len(pValues) == 3 && pValues[2] == "debug" {
+			debug = true
+		}
+		tasks[name] = providerTask(fmt.Sprintf("%s/config/default", path), fmt.Sprintf("%s.provider.yaml", name), debug)
+	}
+
+	return runTaskGroup("resources", tasks)
+}
+
+type taskFunction func(prefix string, errors chan error)
+
+// runTaskGroup executes a group of task in parallel handling an error channel.
+func runTaskGroup(name string, tasks map[string]taskFunction) error {
+	if len(tasks) == 0 {
+		return nil
+	}
+
+	fmt.Printf("[%s] task group started\n", name)
+	defer func(start time.Time) {
+		fmt.Printf("[%s] task group completed, elapsed: %s\n", name, time.Since(start))
+	}(time.Now())
+
+	// Make channels to pass fatal errors in WaitGroup
+	errors := make(chan error)
+	wgDone := make(chan bool)
+
+	wg := new(sync.WaitGroup)
+	for taskName, taskFunction := range tasks {
+		wg.Add(1)
+		taskPrefix := fmt.Sprintf("%s/%s", name, taskName)
+		go runTask(wg, taskPrefix, taskFunction, errors)
+	}
+
+	go func() {
+		wg.Wait()
+		close(wgDone)
+	}()
+
+	// Wait until either WaitGroup is done or an error is received through the error channel.
+	select {
+	case <-wgDone:
+		// carry on
+		break
+	case err := <-errors:
+		close(errors)
```

Comment:
If you do have multiple errors, closing the errors channel might let those other goroutine panic